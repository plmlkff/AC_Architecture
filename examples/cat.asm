LEN: word 0
START:
    ld 1023
    wr LEN
    cmp #0
    je #end
loop:
    ld 1023
    wr 1022
    ld LEN
    sub #1
    je #end
    wr LEN
    jmp #loop
end:
    hlt