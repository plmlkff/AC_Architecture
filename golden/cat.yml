in_source: |
  LEN: word 0
  START:
      ld 1023
      wr LEN
      cmp #0
      je #end
  loop:
      ld 1023
      wr 1022
      ld LEN
      sub #1
      je #end
      wr LEN
      jmp #loop
  end:
      hlt
in_stdin: |-
  5Hello
out_code: |-
  [0, {"name": "ld", "op_code": "LD", "addressing_type": "ABSOLUTE_STRAIGHT", "is_start": true, "address": 1023}, {"name": "wr", "op_code": "WR", "addressing_type": "ABSOLUTE_STRAIGHT", "is_start": false, "address": 0}, {"name": "cmp", "op_code": "CMP", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 0}, {"name": "je", "op_code": "JE", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 12}, {"name": "ld", "op_code": "LD", "addressing_type": "ABSOLUTE_STRAIGHT", "is_start": false, "address": 1023}, {"name": "wr", "op_code": "WR", "addressing_type": "ABSOLUTE_STRAIGHT", "is_start": false, "address": 1022}, {"name": "ld", "op_code": "LD", "addressing_type": "ABSOLUTE_STRAIGHT", "is_start": false, "address": 0}, {"name": "sub", "op_code": "SUB", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 1}, {"name": "je", "op_code": "JE", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 12}, {"name": "wr", "op_code": "WR", "addressing_type": "ABSOLUTE_STRAIGHT", "is_start": false, "address": 0}, {"name": "jmp", "op_code": "JMP", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 5}, {"name": "hlt", "op_code": "HLT", "addressing_type": "NO_ADDRESS", "is_start": false, "address": 0}, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
out_stdout: |
  code.o
  ============================================================
  Hello
  Ticks count: 1143
out_log: |
  DEBUG   Machine:tick_log      Tick #1105:  mIP: 22; AC: 1; BR: 0; DR: 1; SP: -3; CR: Command(name='sub', op_code=<OpCode.SUB: 22>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=1); IP: 8; AR: 8; N: False; Z: False
  DEBUG   Machine:tick_log      Tick #1106:  mIP: 23; AC: 0; BR: 0; DR: 1; SP: -3; CR: Command(name='sub', op_code=<OpCode.SUB: 22>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=1); IP: 8; AR: 8; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1107:  mIP: 69; AC: 0; BR: 0; DR: 1; SP: -3; CR: Command(name='sub', op_code=<OpCode.SUB: 22>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=1); IP: 8; AR: 8; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1108:  mIP: 70; AC: 0; BR: 0; DR: 1; SP: -3; CR: Command(name='sub', op_code=<OpCode.SUB: 22>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=1); IP: 9; AR: 8; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1109:  mIP: 0; AC: 0; BR: 0; DR: 1; SP: -3; CR: Command(name='sub', op_code=<OpCode.SUB: 22>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=1); IP: 9; AR: 8; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1110:  mIP: 1; AC: 0; BR: 0; DR: 1; SP: -3; CR: Command(name='sub', op_code=<OpCode.SUB: 22>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=1); IP: 9; AR: 9; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1111: MEMORY ACCESS TICK mIP: 1; AC: 0; BR: 0; DR: 1; SP: -3; CR: Command(name='sub', op_code=<OpCode.SUB: 22>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=1); IP: 9; AR: 9; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1112: MEMORY ACCESS TICK mIP: 1; AC: 0; BR: 0; DR: 1; SP: -3; CR: Command(name='sub', op_code=<OpCode.SUB: 22>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=1); IP: 9; AR: 9; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1113: MEMORY ACCESS TICK mIP: 1; AC: 0; BR: 0; DR: 1; SP: -3; CR: Command(name='sub', op_code=<OpCode.SUB: 22>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=1); IP: 9; AR: 9; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1114: MEMORY ACCESS TICK mIP: 1; AC: 0; BR: 0; DR: 1; SP: -3; CR: Command(name='sub', op_code=<OpCode.SUB: 22>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=1); IP: 9; AR: 9; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1115: MEMORY ACCESS TICK mIP: 1; AC: 0; BR: 0; DR: 1; SP: -3; CR: Command(name='sub', op_code=<OpCode.SUB: 22>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=1); IP: 9; AR: 9; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1116: MEMORY ACCESS TICK mIP: 1; AC: 0; BR: 0; DR: 1; SP: -3; CR: Command(name='sub', op_code=<OpCode.SUB: 22>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=1); IP: 9; AR: 9; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1117: MEMORY ACCESS TICK mIP: 1; AC: 0; BR: 0; DR: 1; SP: -3; CR: Command(name='sub', op_code=<OpCode.SUB: 22>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=1); IP: 9; AR: 9; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1118: MEMORY ACCESS TICK mIP: 1; AC: 0; BR: 0; DR: 1; SP: -3; CR: Command(name='sub', op_code=<OpCode.SUB: 22>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=1); IP: 9; AR: 9; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1119: MEMORY ACCESS TICK mIP: 1; AC: 0; BR: 0; DR: 1; SP: -3; CR: Command(name='sub', op_code=<OpCode.SUB: 22>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=1); IP: 9; AR: 9; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1120: MEMORY ACCESS TICK mIP: 1; AC: 0; BR: 0; DR: 1; SP: -3; CR: Command(name='sub', op_code=<OpCode.SUB: 22>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=1); IP: 9; AR: 9; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1121:  mIP: 2; AC: 0; BR: 0; DR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=12); SP: -3; CR: Command(name='sub', op_code=<OpCode.SUB: 22>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=1); IP: 9; AR: 9; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1122:  mIP: 3; AC: 0; BR: 0; DR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=12); SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=12); IP: 9; AR: 9; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1123:  mIP: 16; AC: 0; BR: 0; DR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=12); SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=12); IP: 9; AR: 9; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1124:  mIP: 17; AC: 0; BR: 0; DR: 12; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=12); IP: 9; AR: 9; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1125:  mIP: 56; AC: 0; BR: 0; DR: 12; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=12); IP: 9; AR: 9; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1126:  mIP: 57; AC: 0; BR: 0; DR: 12; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=12); IP: 9; AR: 9; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1127:  mIP: 58; AC: 0; BR: 0; DR: 12; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=12); IP: 12; AR: 9; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1128:  mIP: 0; AC: 0; BR: 0; DR: 12; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=12); IP: 12; AR: 9; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1129:  mIP: 1; AC: 0; BR: 0; DR: 12; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=12); IP: 12; AR: 12; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1130: MEMORY ACCESS TICK mIP: 1; AC: 0; BR: 0; DR: 12; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=12); IP: 12; AR: 12; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1131: MEMORY ACCESS TICK mIP: 1; AC: 0; BR: 0; DR: 12; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=12); IP: 12; AR: 12; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1132: MEMORY ACCESS TICK mIP: 1; AC: 0; BR: 0; DR: 12; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=12); IP: 12; AR: 12; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1133: MEMORY ACCESS TICK mIP: 1; AC: 0; BR: 0; DR: 12; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=12); IP: 12; AR: 12; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1134: MEMORY ACCESS TICK mIP: 1; AC: 0; BR: 0; DR: 12; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=12); IP: 12; AR: 12; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1135: MEMORY ACCESS TICK mIP: 1; AC: 0; BR: 0; DR: 12; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=12); IP: 12; AR: 12; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1136: MEMORY ACCESS TICK mIP: 1; AC: 0; BR: 0; DR: 12; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=12); IP: 12; AR: 12; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1137: MEMORY ACCESS TICK mIP: 1; AC: 0; BR: 0; DR: 12; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=12); IP: 12; AR: 12; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1138: MEMORY ACCESS TICK mIP: 1; AC: 0; BR: 0; DR: 12; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=12); IP: 12; AR: 12; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1139: MEMORY ACCESS TICK mIP: 1; AC: 0; BR: 0; DR: 12; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=12); IP: 12; AR: 12; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1140:  mIP: 2; AC: 0; BR: 0; DR: Command(name='hlt', op_code=<OpCode.HLT: 68>, addressing_type=<AddressingType.NO_ADDRESS: 5>, is_start=False, address=0); SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=12); IP: 12; AR: 12; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1141:  mIP: 3; AC: 0; BR: 0; DR: Command(name='hlt', op_code=<OpCode.HLT: 68>, addressing_type=<AddressingType.NO_ADDRESS: 5>, is_start=False, address=0); SP: -3; CR: Command(name='hlt', op_code=<OpCode.HLT: 68>, addressing_type=<AddressingType.NO_ADDRESS: 5>, is_start=False, address=0); IP: 12; AR: 12; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1142:  mIP: 19; AC: 0; BR: 0; DR: Command(name='hlt', op_code=<OpCode.HLT: 68>, addressing_type=<AddressingType.NO_ADDRESS: 5>, is_start=False, address=0); SP: -3; CR: Command(name='hlt', op_code=<OpCode.HLT: 68>, addressing_type=<AddressingType.NO_ADDRESS: 5>, is_start=False, address=0); IP: 12; AR: 12; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #1143:  mIP: 68; AC: 0; BR: 0; DR: Command(name='hlt', op_code=<OpCode.HLT: 68>, addressing_type=<AddressingType.NO_ADDRESS: 5>, is_start=False, address=0); SP: -3; CR: Command(name='hlt', op_code=<OpCode.HLT: 68>, addressing_type=<AddressingType.NO_ADDRESS: 5>, is_start=False, address=0); IP: 12; AR: 12; N: False; Z: True
